### PR TITLE
improve(liquidator): Self audit liquidator's WDF strategy implementation

### LINKED ISSUE #2249: Self-audit use of WDF strategy in liquidator bot
**Why**

#2203 fixed how `Liquidatable.sol` contract resets withdrawal liveness expirations and liquidator bot unit tests that were giving false positives were fixed as well. This issue calls for a subsequent audit of the liquidator bot strategy to make sure that it is taking advantage of the withdrawal reset feature correctly.

It is possible errors were made in the original implementation of the WDF because the implementer was referencing a bugged smart contract.

**Difficulty Score [1-10]**

This is a measure of the amount of developer time this task will require to be completed. Note: this should include design and planning time in addition to implementation time. The scale is not linear:

- 4: ~2 hours

### PR DESCRIPTION
**Pull request by nicholaspai**
Signed-off-by: Nick Pai <npai.nyc@gmail.com>

**Motivation**

#2249 

**Summary**

Potential other tasks:
- Improve documentation with examples
- Recommend default values for WDF-related params
- More unit tests demonstrating more edge cases

**Issue(s)**

Fixes #2249

### CODE REVIEW

File: packages/liquidator/test/Liquidator.js [hunk 1]
```diff
@@ -1044,6 +1049,12 @@ contract("Liquidator.js", function(accounts) {
           // This places their position with a CR of: 115 / (100 * 1) * 100 = 115%. This is below the CR threshold.
           await emp.requestWithdrawal({ rawValue: convert("10") }, { from: sponsor1 });
 
+          // Advance time to the defenseActivationPercent to see if the WDF would trigger.
+          let sponsor1Positions = await emp.positions(sponsor1);
+          const withdrawLiveness = empProps.withdrawLiveness.toNumber();
+          let nextTime = Math.ceil(Number(sponsor1Positions.withdrawalRequestPassTimestamp) - withdrawLiveness * 0.5);
+          await emp.setCurrentTime(nextTime);
```
Comment (daywiss) [inline packages/liquidator/test/Liquidator.js:1056]: nice catch here, yea this test probably would not have worked as intended

**Review by daywiss — APPROVED** [review]
looks good

**Comment by daywiss** [comment]
sorry did not realize this was in draft

File: packages/liquidator/test/Liquidator.js (hunk 1 above)
Comment (nicholaspai) [reply packages/liquidator/test/Liquidator.js:1056]: Yeah well it works but its a big of a false positive

**Comment by nicholaspai** [comment]
> sorry did not realize this was in draft

Thanks for initial review though, which I requested.

File: packages/liquidator/test/Liquidator.js (hunk 1 above)
Comment (nicholaspai) [reply packages/liquidator/test/Liquidator.js:1056]: For others: without this line here, the WDF would not trigger even if the bot couldn't fully liquidate the position because not enough time has passed (e.g. `defenseActivationPercent`).

**Review by mrice32 — APPROVED** [review]
This is awesome! Nice find @nicholaspai

**Review by daywiss — APPROVED** [review]
